### PR TITLE
[refactor][cli][PIP-280] Refactor  `pulsar-client-tools` module

### PR DESCRIPTION
**Pull request by JooHyukKim**
PIP: #20691 

### Motivation

- In the current Pulsar codebase, the logic to parse CLI arguments for measurement units like time and bytes is scattered across various CLI classes. Each value read has its distinct parsing implementation, leading to a lack of code reuse.

### Modifications

- Implement validators and converters to replace manual parsing logic in Cmd classes
- Apply refactor to...
    -  CmdNamespaces
    - CmdPersistetTopics
    - CmdTopicPolicies
    - CmdTopics
    - TimeUnitTo

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/JooHyukKim/pulsar/pull/18

### CODE REVIEW

**Comment by tisonkun** [comment]
Thanks for your contribution! I'll give a review this weekend.

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java [hunk 1]
```diff
@@ -747,39 +743,27 @@ private class SetRetention extends CliCommand {
...
-            final int retentionSizeInMB;
-            if (sizeLimit != -1) {
-                retentionSizeInMB = (int) (sizeLimit / (1024 * 1024));
-            } else {
-                retentionSizeInMB = -1;
-            }
+            final int retentionTimeInMin = retentionTimeInSec !=  -1
+                    ? (int) TimeUnit.SECONDS.toMinutes(retentionTimeInSec)
+                    : retentionTimeInSec.intValue();
+            final int retentionSizeInMB = sizeLimit != -1
```
Comment (tisonkun) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java:764]: Perhaps we can later move this size conversion to `ByteUnitUtil` also.

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java [hunk 2]
```diff
@@ -1311,23 +1297,16 @@ void run() throws PulsarAdminException {
             BacklogQuota.Builder builder = BacklogQuota.builder().retentionPolicy(policy);
             if (backlogQuotaType == BacklogQuota.BacklogQuotaType.destination_storage) {
                 // set quota by storage size
-                if (limitStr == null) {
+                if (limit == null) {
```
Comment (tisonkun) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java:1300]: We may use a Nonnull ValueValidator here?

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 3]
```diff
@@ -2225,62 +2178,7 @@ void run() throws PulsarAdminException {
                     + " if s3 offload enabled");
             }
 
-            int maxBlockSizeInBytes = OffloadPoliciesImpl.DEFAULT_MAX_BLOCK_SIZE_IN_BYTES;
-            if (StringUtils.isNotEmpty(maxBlockSizeStr)) {
-                long maxBlockSize = validateSizeString(maxBlockSizeStr);
-                if (positiveCheck("MaxBlockSize", maxBlockSize)
```
Comment (tisonkun) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:2231]: We can remove `positiveCheck` and `maxValueCheck` utils in this class now.

**Review by tisonkun — COMMENTED** [review]
Thanks for your contribution @JooHyukKim!

This patch is in a good direction. Comments inline. And please check the CI failure.

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java (hunk 2 above)
Comment (tisonkun) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java:1300]: Thanks for your explanation! Let's keep it AS IS then.

**Review by tisonkun — APPROVED** [review]
LGTM. We should try to fix the CI issue.

**Comment by tisonkun** [comment]
@JooHyukKim Please read the CI output before retry, it gives:

```
pulsar-cli-utils-3.2.0-SNAPSHOT.jar unaccounted for in LICENSE
```

So you may need to update some LICENSE file. Please check the CI output for details.

**Comment by tisonkun** [comment]
Rebase on #20663. Let's test once more.

**Comment by tisonkun** [comment]
Thanks for your contributon! The first migration step is merged now :D

**Comment by JooHyukKim** [comment]
@tisonkun Thank you also for considerate time investment on this one! I will try to pick up where I left off 👍🏼.

**Pull request merged**